__tests__/components/MessageDisplay.spec.ts

```diff
@@ -0,0 +1,272 @@
+                namespaced: true,
+                getters: {
+                    currentAccount: () => WalletsModel1,
+                    currentRecipient: () => recipientPublicAccount,
+                },
+            },
+            namespace: {
+                namespaced: true,
+                getters: {
+                    linkedAddress: () => linkedAddress,
+                },
+            },
+        };
+
+        return getComponent(MessageDisplay, mockStore, {}, props, {}, dispatch);
+    };
+
+    describe('loadDetails()', () => {
+        const runLoadDetailsTest = async (
+            unannounced: boolean,
+            message: PlainMessage | EncryptedMessage,
+            expectedMessageDisplay: string,
+        ) => {
+            // Arrange:
+            const props = {
+                message,
+                unannounced,
+            };
+            const mockT = jest.fn().mockImplementation((_) => _);
+
+            // Act:
+            const wrapper = getMessageDisplayWrapper(props);
+            const vm = wrapper.vm as MessageDisplayTs;
+            vm['$t'] = mockT;
+            vm['loadDetails']();
+            const messageDisplay = vm['messageDisplay'];
+
+            // Assert:
+            expect(messageDisplay).toBe(expectedMessageDisplay);
+        };
+
+        test('message is encrypted and tx is unannounced', async () => {
+            // Arrange:
+            const unannounced = true;
+            const message = EncryptedMessage.create('message', account1.publicAccount, account1.privateKey);
+            const expectedMessageDisplay = `${message.payload} (encrypted_message)`;
+
+            // Act + Assert:
+            await runLoadDetailsTest(unannounced, message, expectedMessageDisplay);
+        });
+
+        test('message is encrypted and tx is announced', async () => {
+            // Arrange:
+            const unannounced = false;
+            const message = EncryptedMessage.create('message', account1.publicAccount, account1.privateKey);
+            const expectedMessageDisplay = '******';
+
+            // Act + Assert:
+            await runLoadDetailsTest(unannounced, message, expectedMessageDisplay);
+        });
+
+        test('message is unencrypted', async () => {
+            // Arrange:
+            const unannounced = false;
+            const message = PlainMessage.create('message');
+            const expectedMessageDisplay = 'message';
+
+            // Act + Assert:
+            await runLoadDetailsTest(unannounced, message, expectedMessageDisplay);
+        });
+    });
+
+    describe('onAccountUnlocked()', () => {
+        const runUnlockAccountTest = async (
+            recipient: UnresolvedAddress,
+            linkedAddress: Address | null,
+            decryptAddress: Address | null,
+            expectGetLinkedAddressToBeDispatched: boolean,
+            expectAddErrorToBeDispatched: boolean,
+            expecDecryptMessageToBeCalled: boolean,
+        ) => {
```

Comment:
consider  passing in the options via a single object. that might lead to improved readability.

Reply:
i was thinking about having the calling code look something like this:

```js
            await runUnlockAccountTest(recipient, linkedAddress, decryptAddress, {
                getLinkedAddressToBeDispatched: true,
                addErrorToBeDispatched: false,
                linkedAddressToBeDispatched: true
            });
```
the first three are required and outside of the object. the latter three are part of an object (and self describing since keys are specified). in that case, the parameter can be called a more descriptive "expectations" instead of "options".

Reply:
Should we move the definition of `expectations` under `Arrange` section?

```js
// Arrange:
...
const expectations = {
    getLinkedAddressToBeDispatched: true,
    addErrorToBeDispatched: false,
    decryptMessageToBeCalled: true
}

// Act + Assert:
await runUnlockAccountTest(recipient, linkedAddress, decryptAddress, expectations);
```

Reply:
i don't mind having it inline, but your choice